src/test/java/uk/gov/hmcts/reform/ccd/service/CaseFamilyTreeServiceTest.java

```diff
@@ -323,29 +324,105 @@ void testGetCaseFamilies() {
 
-                assertThat(caseFamily2.getRootAncestor().getId()).isEqualTo(2L);
+                assertThat(caseFamily2.getRootCase().getId()).isEqualTo(2L);
                 assertThat(familyMembers2).hasSize(3).hasSameElementsAs(List.of(10L, 11L, 100L));
 
-                assertThat(caseFamily3.getRootAncestor().getId()).isEqualTo(1000L);
+                assertThat(caseFamily3.getRootCase().getId()).isEqualTo(1000L);
                 assertThat(familyMembers3).isEmpty();
             });
     }
 
+    @Test
+    @DisplayName("Get case families when cases are cyclically linked.")
+    void testGetCaseFamiliesScenario2() {
+        final List<CaseDataEntity> expiredCases = List.of(DELETABLE_CASE_ENTITY_WITH_PAST_TTL, LINKED_CASE_ENTITY_10);
+
+        final CaseLinkEntity caseLink1_10 = new CaseLinkEntityBuilder(1L, DELETABLE_CASE_TYPE, 10L)
+            .build();
+        final CaseLinkEntity caseLink10_1 = new CaseLinkEntityBuilder(10L, DELETABLE_CASE_TYPE, 1L)
+            .build();
+
+        doReturn(List.of(TestData.DELETABLE_CASE_TYPE)).when(parameterResolver).getDeletableCaseTypes();
+        doReturn(expiredCases).when(caseDataRepository).findExpiredCases(anyList());
+        doReturn(List.of(caseLink10_1)).when(caseLinkRepository).findByLinkedCaseId(1L);
+        doReturn(List.of(caseLink1_10)).when(caseLinkRepository).findByLinkedCaseId(10L);
+
+        doReturn(Optional.of(DELETABLE_CASE_ENTITY_WITH_PAST_TTL)).when(caseDataRepository).findById(1L);
+        doReturn(Optional.of(LINKED_CASE_ENTITY_10)).when(caseDataRepository).findById(10L);
+
+        doReturn(List.of(caseLink1_10)).when(caseLinkRepository).findByCaseId(1L);
+        doReturn(List.of(caseLink10_1)).when(caseLinkRepository).findByCaseId(10L);
+
+        final List<CaseFamily> caseFamilies = underTest.getCaseFamilies();
+
+        assertThat(caseFamilies)
+            .isNotEmpty()
+            .hasSize(2)
+            .satisfies(items -> {
+                final CaseFamily caseFamily1 = items.get(0);
+                final List<Long> familyMembers1 = caseFamily1.getLinkedCases().stream()
+                    .map(CaseData::getId)
+                    .collect(Collectors.toUnmodifiableList());
+
+                final CaseFamily caseFamily2 = items.get(1);
+                final List<Long> familyMembers2 = caseFamily2.getLinkedCases().stream()
+                    .map(CaseData::getId)
+                    .collect(Collectors.toUnmodifiableList());
+
+                assertThat(caseFamily1.getRootCase().getId()).isEqualTo(1L);
+                assertThat(familyMembers1).singleElement().isEqualTo(10L);
+
+                assertThat(caseFamily2.getRootCase().getId()).isEqualTo(10L);
+                assertThat(familyMembers2).singleElement().isEqualTo(1L);
+            });
+    }
+
+    @Test
+    @DisplayName("Get case families when cases are cyclically linked.")
+    void testGetCaseFamiliesScenario3() {
```

Comment:
Can you update name and/or description to distinguish from test above.

Reply:
Fixed by commit 910f99ae55f163bb30f100e0e585aeafedd99c35